js/script.js

```diff
@@ -0,0 +1,110 @@
+class App extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            searchText: '',
+            messageText: '',
+            users: [],
+            result: 1
+        };
+    }
+  
+    onChangeHandle(event) {
+        this.setState({searchText: event.target.value});
+    }
+
+    onError(error) {
+        switch(this.state.result) {
+            case '-1':
+                this.setState({
+                    messageText: `No matching users found.`,
+                    users: []
+                })
+                break;
+            default:
+                this.setState({
+                    messageText: `An error has occured. Server error status code: ${error.message}. Application status code: ${this.state.result}.`,
+                    users: [],
+                })
+                console.error(`Error: ${error.message}`);
+                break;
+        }
+    }
+
+    onSubmit(event) {
+        event.preventDefault();
+        const {searchText} = this.state;
+        const url = `https://api.github.com/search/users?q=${searchText}`;
+        fetch(url)
+            .then(response => {
+                if (!response.ok) {
+                    throw Error(response.status);
```

Comment:
Po co jest ten throw? Jak call się nie uda to powinieneś wpaść w `catch`a z 54 linijki